DOC-2735 - Refresh of VM Migration Assistant documentation (#10396)
spectrocloud/librarium: docs/docs-content/vm-management/vm-migration-assistant/additional-configuration.md

tags: ["vmo", "vm migration assistant"]
---

This guide explains the additional configuration options available in the VM Migration Assistant, including how to
adjust settings for your migration controller, providers, and migration plans.

## Overview

> The Overview page of the VM Migration Assistant provides additional tuning and configuration options for your migration

Reviewer: ```suggestion
The **Overview** page of the VM Migration Assistant provides additional tuning and configuration options for your migration
```

controller. It also displays metrics for your migrations and health of the operational pods used for migration.

![VM Migration Assistant - Overview Page](/vm-migration-assistant/additional-configuration_overview-page.webp)

### Overview Tab

View the overall status of your VM Migration Assistant on the **Overview** tab. This includes metrics for your

> migrations and health of the operational pods used for migration.

Reviewer: ```suggestion
migrations and the health of the operational pods used for migration.
```

### YAML Tab

> The **YAML** tab displays a YAML editor for the VM Migration Assistant (`ForkliftController` resource).

Reviewer: ```suggestion
The **YAML** tab displays a YAML code editor for the VM Migration Assistant (`ForkliftController` resource).
```

Per UI meeting.

View and edit the
[Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) YAML on this
tab.

### Health Tab

View the health status and logs for pods used by the VM Migration Assistant on the **Health** tab.

### History Tab

View the history of migrations initiated by the VM Migration Assistant on the **History** tab.

### Settings Tab

The **Settings** tab has configurable settings for the VM Migration Assistant. These parameters define the resource
allocation, performance tuning, and operational settings for efficiently managing virtual machine migrations.

Perform the following steps to change a setting.

1. Click the **Edit** option next to **Settings**.

> 2. Adjust each setting as desired in the pop-up window. The configurable parameters are described in the following

Reviewer: ```suggestion
2. Adjust each setting as desired in the modal window. The configurable parameters are described in the following
```

   | **Maximum concurrent VM migrations**            | The maximum number of VMs that can be migrated simultaneously. Consider the available system resources and network bandwidth, as too many concurrent migrations can overload the system and cause slower speeds or congestion. Start with a lower value and gradually increase it to find the optimal balance. Migration plans can enter an **Unknown** state if you have selected to migrate more VMs than this setting allows. Refer to [Scenario - Virtual Machine (VM) Migrations Plans in Unknown State](../../troubleshooting/vmo-issues.md#scenario---virtual-machine-vm-migration-plans-in-unknown-state) in our troubleshooting section for details of this scenario. | `20`          |

>    | **Controller main container CPU limit**         | The CPU limit (in millicpu) allocated to the main container in the `forklift-controller` pod. Environments running a high number of concurrent migrations, or using providers that require frequent reconciliation, may need this limit increased to prevent throttling from slowing migration progress.                                                                                                                                                                                                                                                                                                                                                                       | `500m`        |

Reviewer: <img width="800" height="600" alt="Image" src="https://github.com/user-attachments/assets/bfd5f38c-828b-45f2-a36b-c5df1afee097" />

Author: 🤣  Only you and I will get this!

   | **Controller inventory container memory limit** | The memory limit (in mebibytes) allocated to the inventory container in the `forklift-controller` pod. The inventory container's memory demand scales with the number of VMs and providers being tracked, so large environments may need this limit raised above the default.                                                                                                                                                                                                                                                                                                                                                                                                  | `1000Mi`      |

>    | **Precopy interval**                            | _(Only applicable to warm migrations)_ The interval (in minutes) at which new Changed Block Tracking (CBT) snapshots are requested and transferred throughout the precopy stage. Shorter intervals reduce the amount of data transferred at cutover, but consume the snapshot limit faster, which can cause warm migration to fail if cutover is not triggered in time.                                                                                                                                                                                                                                                                                                        | `60min`       |

Reviewer: ```suggestion
   | **Precopy interval**                            | _(Only applicable to warm migrations)_ The interval (in minutes) at which new Changed Block Tracking (CBT) snapshots are requested and transferred throughout the precopy stage. Shorter intervals reduce the amount of data transferred at cutover but consume the snapshot limit faster, which can cause warm migrations to fail if cutover is not triggered in time.                                                                                                                                                                                                                                                                                                        | `60min`       |
```

>    | **Snapshot polling interval**                   | _(Only applicable to warm migrations)_ The frequency at which the migration controller polls to check whether a snapshot creation or removal operation has completed. A shorter interval means the controller detects completed snapshots sooner and can proceed to the next step faster, at the cost of slightly increased API call frequency against the source provider.                                                                                                                                                                                                                                                                                                    | `10s`         |

Reviewer: ```suggestion
   | **Snapshot polling interval**                   | _(Only applicable to warm migrations)_ The frequency at which the migration controller polls check whether a snapshot creation or removal operation has completed. A shorter interval means the controller detects completed snapshots sooner and can proceed to the next step faster, at the cost of slightly increased API call frequency against the source provider.                                                                                                                                                                                                                                                                                                    | `10s`         |
```

3. Click **Save** after making changes.

## Provider Settings

Select **Providers** on the left main menu to view your source and host providers.

Click on a provider name to view its settings.

![Providers Page](/vm-migration-assistant/additional-configuration_providers-page.webp)

### Details Tab

<Tabs queryId="provider-details-tabs">

<TabItem value="source-provider" label="Source Provider">

Perform the following steps to change a setting.

1. Click the pencil icon next to each value.

> 2. Adjust the value in the pop-up window.

Reviewer: ```suggestion
2. Adjust the value in the modal window.
```

   The configurable settings are described in the following table. These were originally defined when you performed the

>    steps in [Create Source Providers](./create-source-providers.md).

Reviewer: ```suggestion
   The configurable settings are described in the following table. These are defined when you perform the
   steps in [Create Source Providers](./create-source-providers.md).
```

   | ------------------------ | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
   | **URL**                  | Your vSphere / ESXi API endpoint for the software development kit (SDK). You can specify a Fully Qualified Domain Name (FQDN) or an IP address. For example, `https://vcenter.mycompany.com/sdk`.                                                                                                                                                               |
   | **VDDK init image**      | Provide the registry URL to the VMware Virtual Disk Development Kit (VDDK) image. Specify the value without the HTTP scheme `https://` or `http://`. For example, `docker.io/myorganization/vddk:v8.0.3`.                                                                                                                                                       |

>    | **External web UI link** | Your vSphere / ESXi UI endpoint. You can specify a Fully Qualified Domain Name (FQDN) or an IP address. For example, `https://vcenter.mycompany.com/ui`.                                                                                                                                                                                                        |

Reviewer: ```suggestion
   | **External web UI link** | Your vSphere / ESXi UI endpoint. You can specify an FQDN or an IP address. For example, `https://vcenter.mycompany.com/ui`.                                                                                                                                                                                                        |
```

   | **Convert Disk**         | When enabled, disk conversion is handled using `virt-v2v`. For example, if you're migrating from VMware to Virtual Machine Orchestrator (VMO), `virt-v2v` can convert Virtual Machine Disk (VMDK) to raw or QEMU copy-on-write version 2 (qcow2) formats that are optimal for the target environment. When disabled, disks are transferred using raw copy mode. |

3. Click **Save** after making changes.

</TabItem>

<TabItem value="host-provider" label="Host Provider">

Perform the following steps to change a setting.

1. Click the pencil icon next to each value.

> 2. Adjust the value in the pop-up window.

Reviewer: ```suggestion
2. Adjust the value in the modal window.
```

   The configurable settings are described in the following table.

   | Setting                      | Description                                                                                                                                                                                                                                                               |
   | ---------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

>    | **Default Transfer Network** | _Use the [Settings Tab](#settings-tab) on the [Overview](#overview) page to adjust this setting at the controller level. This prevents changes from being overridden._ Select a default migration network. If no network is selected, the pod network is used by default. |

Reviewer: ```suggestion
   | **Default Transfer Network** |  _Use the [Settings Tab](#settings-tab) on the [Overview](#overview) page to adjust this setting at the controller level. This prevents changes from being overridden._ <br /> <br /> Select a default migration network. If no network is selected, the pod network is used by default.  |
```

Aligning with how you did the others later on, with a blank line in between.

3. Click **Save** after making changes.

</TabItem>

</Tabs>

### YAML Tab

> The **YAML** tab displays a YAML editor for the provider resource.

Reviewer: ```suggestion
The **YAML** tab displays a YAML code editor for the provider resource.
```

View and edit the
[Custom Resource](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) YAML on this
tab.

### Credentials Tab

Perform the following steps to edit the provider credentials.

1. Click the **Edit** option next to **Credentials**.

> 2. Update the value for each setting that you want to change in the pop-up window.

Reviewer: ```suggestion
2. In the modal window, update the value for each setting that you want to change.
```

   The configurable settings are described in the following table. These were originally defined when you performed the

>    steps in [Create Source Providers](./create-source-providers.md).

Reviewer: ```suggestion
   The configurable settings are described in the following table. These are defined when you perform the
   steps in [Create Source Providers](./create-source-providers.md).
```

   | Setting                              | Description                                                                                               |
   | ------------------------------------ | --------------------------------------------------------------------------------------------------------- |
   | **Username**                         | Your vSphere / ESXi account username. For example, `user@vsphere.local`.                                  |
   | **Password**                         | Your vSphere / ESXi account password.                                                                     |
   | **Certificate validation**           | **Configure certificate validation** for your vSphere / ESXi provider or **Skip certificate validation**. |

>    | **Configure certificate validation** | Upload or drag and drop the CA certificate for your vSphere / ESXi.                                       |

Reviewer: ```suggestion
   | **Configure certificate validation** | Upload or drag and drop the certificate authority (CA) certificate for your vSphere / ESXi.                                       |
```

[NetworkAttachmentDefinitions](https://docs.openshift.com/container-platform/4.8/rest_api/network_apis/networkattachmentdefinition-k8s-cni-cncf-io-v1.html)
from the cluster. This tab is only visible on host clusters.

Use the following steps to select a default migration network for the cluster to improve disk transfer performance. If
no network is selected, the pod network is used by default, which may not be optimal.

1. Click on **Set provider default transfer network**.

2. Select a network from the **Default transfer Network** drop-down.

3. After making changes, click **Save**.

## Plan Settings

Select **Plans** on the left main menu to view your migration plans.

Click on a migration plan name to view its settings.

![Migration Plans Page](/vm-migration-assistant/additional-configuration_migration-plans-page.webp)

:::info

Settings cannot be changed for migration plans that are in progress or complete.

:::

### Details Tab

Perform the following steps to change a setting.

1. Click the pencil icon next to each value.

> 2. Adjust the value in the pop-up window.

Reviewer: ```suggestion
2. Adjust the value in the modal window.
```

   The configurable settings are described in the following table. Some of these were originally defined when you

>    performed the steps in [Create Migration Plans](./create-migration-plans.md).

Reviewer: ```suggestion
   The configurable settings are described in the following table. Some of these are defined when you
   perform the steps in [Create Migration Plans](./create-migration-plans.md).
```

   #### Defined at Plan Creation

   | Setting                 | Description                                                                                                                                                                                                                                                                                                              |
   | ----------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
   | **Description**         | An optional description for your migration plan.                                                                                                                                                                                                                                                                         |

>    | **Migration type**      | Choose whether this will be a warm or cold migration. A cold migration is when VMs are shut down at the start of migration. A warm migration is when active VMs are migrated while they are running, and are shut down during the final switchover.                                                                      |

Reviewer: ```suggestion
   | **Migration type**      | Choose whether this will be a warm or cold migration. A cold migration is when VMs are shut down at the start of migration. A warm migration is when active VMs are migrated while they are running and are shut down during the final switchover.                                                                      |
```

   #### Additional Configurable Settings

   | Setting                      | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
   | ---------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |

>    | **Guest conversion mode**    | _This is defined at the Provider level with the **Convert disk** option, but can be overridden at the Plan level by using this setting._ <br /><br /> When enabled, disk conversion is handled using `virt-v2v`. For example, if you're migrating from VMware to Virtual Machine Orchestrator (VMO), `virt-v2v` can convert Virtual Machine Disk (VMDK) to raw or QEMU copy-on-write version 2 (qcow2) formats that are optimal for the target environment. When disabled, disks are transferred using raw copy mode. |

Reviewer: ```suggestion
   | **Guest conversion mode**    | _This is defined at the Provider level with the **Convert disk** option, but can be overridden at the Plan level by using this setting._ <br /><br /> When enabled, disk conversion is handled using `virt-v2v`. For example, if you are migrating from VMware to VMO, `virt-v2v` can convert VMDK to raw or QEMU copy-on-write version 2 (qcow2) formats that are optimal for the target environment. When disabled, disks are transferred using raw copy mode. |
```

Author: Somehow, I always sneak a contraction in...

#### Remove VMs from Plan

:::info

> If you only have one VM remaining in your plan, delete the plan instead, as deleting all VMs from a plan is not allowed.

Reviewer: ```suggestion
If you only have one VM remaining in your plan, you must delete the plan instead. Deleting all VMs from a plan is not allowed.
```

:::

Perform the following steps to remove VMs from your plan. You cannot remove VMs from a plan if it is already in progress
or complete.

1. Click the checkbox next to each VM that you want to remove from the plan.

2. Click on **Delete virtual machines**.

> 3. Click **Delete** in the pop-up window.

Reviewer: ```suggestion
3. Click **Delete** in the dialog window.
```

### Resources Tab

> The **Resources** tab displays the the total amount of CPU and memory resources that all VMs in the migration plan will

Author: ```suggestion
The **Resources** tab displays the total amount of CPU and memory resources that all VMs in the migration plan will
```

require in the target environment.

### Mappings tab

The **Mappings** tab displays the network and storage mappings for the migration plan. These were originally defined

> when you performed the steps in [Create Migration Plans](./create-migration-plans.md).

Reviewer: ```suggestion
The **Mappings** tab displays the network and storage mappings for the migration plan. These are defined
when you perform the steps in [Create Migration Plans](./create-migration-plans.md).
```

#### Manage Mappings

1. Click **Edit** next to **Network map** or **Storage map**.

2. Adjust the current mapping by selecting a different option from the drop-downs.

   You can click **Add mapping** to add an additional mapping and delete existing mappings by clicking the minus icon
   next to each mapping.

3. Click **Save** after making changes.

### Hooks Tab

The **Hooks** tab displays the migration hooks for the migration plan. These were originally defined when you performed

> the steps in [Create Migration Plans](./create-migration-plans.md).

Reviewer: ```suggestion
The **Hooks** tab displays the migration hooks for the migration plan. These are defined when you perform
the steps in [Create Migration Plans](./create-migration-plans.md).
```

2. You can enable or disable the hook using the toggle. If you enable the hook, additional settings are displayed as
   described in the following table.

   | Setting               | Description                                                                                                                                                                                           | Example                       |
   | --------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------- |
   | **Hook runner image** | Specify a repository and tag for the hook runner image or custom image. The hook runner is a container that runs your pre-migration and post-migration hooks.                                         | `quay.io/myrepo/hooks:latest` |
   | **Service account**   | Specify a service account for the hook runner to use. The service account must exist in your target namespace.                                                                                        | `hook-runner-service-account` |

>    | **Ansible playbook**  | You can optionally provide an [Ansible playbook](https://ansible.readthedocs.io/projects/runner/en/stable/intro/) for the hook. You can only specify a playbook if you are using a hook-runner image. |                               |

Reviewer: ```suggestion
    | **Ansible playbook**  | You can optionally provide an [Ansible playbook](https://ansible.readthedocs.io/projects/runner/en/stable/intro/) for the hook. You can only specify a playbook if you enter a **Hook runner image**. |         N/A                      |
```

Or a hyphen? Just so it's not blank?
